From koba@kmckk.co.jp Mon Sep 24 22:09:59 2012
From: Tetsuyuki Kobayashi <koba@kmckk.co.jp>
Date: Tue, 25 Sep 2012 14:07:53 +0900
Subject: [PATCH 17/21] ARM: shmobile: fix smp build
To: greg@kroah.com
Cc: ltsi-dev@lists.linuxfoundation.org, horms@verge.net.au, damm@opensource.se, kuninori.morimoto.gx@renesas.com, Arnd Bergmann <arnd@arndb.de>, Tetsuyuki Kobayashi <koba@kmckk.co.jp>
Message-ID: <1348549677-1458-18-git-send-email-koba@kmckk.co.jp>


From: Arnd Bergmann <arnd@arndb.de>

I got build errors with the new version now because machine_is_kzm9g is no longer
defined:

arch/arm/mach-shmobile/platsmp.c: In function 'shmobile_smp_get_core_count':
arch/arm/mach-shmobile/platsmp.c:29:2: error: implicit declaration of function 'of_machine_is_compatible'

Replace the missing function with a call to of_machine_is_compatible.

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Acked-by: "Rafael J. Wysocki" <rjw@sisk.pl>
Acked-by: Magnus Damm <magnus.damm@gmail.com>
(cherry picked from commit 9601e87388f4969a80b021a40215d24c4e78d5b3)

Signed-off-by: Tetsuyuki Kobayashi <koba@kmckk.co.jp>
---
 arch/arm/boot/dts/sh73a0-kzm9g.dts |    2 +-
 arch/arm/mach-shmobile/platsmp.c   |    4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/sh73a0-kzm9g.dts b/arch/arm/boot/dts/sh73a0-kzm9g.dts
index ed1bae5..bcb9119 100644
--- a/arch/arm/boot/dts/sh73a0-kzm9g.dts
+++ b/arch/arm/boot/dts/sh73a0-kzm9g.dts
@@ -13,7 +13,7 @@
 
 / {
 	model = "KZM-A9-GT";
-	compatible = "renesas,kzm9g";
+	compatible = "renesas,kzm9g", "renesas,sh73a0";
 
 	memory {
 		device_type = "memory";
diff --git a/arch/arm/mach-shmobile/platsmp.c b/arch/arm/mach-shmobile/platsmp.c
index d959f07..5a2b69c 100644
--- a/arch/arm/mach-shmobile/platsmp.c
+++ b/arch/arm/mach-shmobile/platsmp.c
@@ -16,11 +16,13 @@
 #include <linux/device.h>
 #include <linux/smp.h>
 #include <linux/io.h>
+#include <linux/of.h>
 #include <asm/hardware/gic.h>
 #include <asm/mach-types.h>
 #include <mach/common.h>
 
-#define is_sh73a0() (machine_is_ag5evm() || machine_is_kota2() || machine_is_kzm9g())
+#define is_sh73a0() (machine_is_ag5evm() || machine_is_kota2() || \
+			of_machine_is_compatible("renesas,sh73a0"))
 #define is_r8a7779() machine_is_marzen()
 
 static unsigned int __init shmobile_smp_get_core_count(void)
-- 
1.7.9.5

